### PR TITLE
Feature: improved non EIP-1559 gas pricing logic

### PR DESCRIPTION
**Pull request by ACC01ADE**
## Describe Changes

Fixing gas price bug that included zero priced transactions.
This improves the gas price calculations greatly for networks like BSC.

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Code styles have been enforced
- [x] I have checked eslint

### CODE REVIEW

File: src/utils/network-monitor.ts [hunk 1]
```diff
@@ -968,10 +968,13 @@ export class NetworkMonitor {
       }
     }
     // for legacy networks, get average gasPrice
-    else if (this.gasPrices[network].gasPrice === null) {
-      this.gasPrices[network].gasPrice = tx.gasPrice!
-    } else {
-      this.gasPrices[network].gasPrice = this.gasPrices[network].gasPrice!.add(tx.gasPrice!).div(TWO)
+    // it's important to skip this calculation if gas price is 0, which happens in some instances like on BSC
+    else if (tx.gasPrice !== null && tx.gasPrice!.gt(ZERO)) {
+      if (this.gasPrices[network].gasPrice === null) {
```
Comment (alexanderattar) [inline src/utils/network-monitor.ts:975]: Can you add a couple comments on the if and else statements here for the reasoning behind the logic?

**Review by alexanderattar — APPROVED** [review]

File: src/utils/network-monitor.ts (hunk 1 above)
Comment (ACC01ADE) [reply src/utils/network-monitor.ts:975]: Provided more commentary in next commit.

**Review by sogoiii — APPROVED** [review]
👍

**Review by sogoiii — APPROVED** [review]
👍

**Pull request merged**